Make validTo field in consentRecord optional

stevenvegt opened the pull request.

stevenvegt committed:
Make validTo optional in spec and generated code

stevenvegt committed:
Add migration for nullable valid_to

stevenvegt committed:
Handle optional validTo

Use pointer value and fix specs.

stevenvegt commented:
@woutslakhorst can you take a good look at my migrations? First time writing them and unsure if its correct. E.g. the `down` migration might not work when there are actual rows with `valid_to = null`.

woutslakhorst commented on migrations/4_alter_consent_record_make_valid_to_optional.down.sql:
```diff
@@ -0,0 +1,20 @@
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    patient_consent_id VARCHAR(255) REFERENCES patient_consent(id),
+    valid_from DATE NOT NULL,
+    valid_to DATE NOT NULL,
```
Add a default value in 2100 or so.

Then the question is if you're allowed to do an INSERT with null value, does it error or does it use default?
If error, the insert needs to be separated into 2 queries: one with valid_to and one without.

stevenvegt replied:
Done

woutslakhorst reviewed (changes requested):
Still some red flags in the build. Coverage declined because of the extra if-statements. Should add some extra tests.

stevenvegt committed:
Fix down migration

Add default date on NULL columns.
Split data copy step into 2 queries, with and without end_date,

stevenvegt committed:
Fix test coverage for httpClient.CreateConsent

Add case with a validTo.

stevenvegt commented:
FYI: Codacy breaks because of generated bin-data file.
